woocommerce/woocommerce
# Update SelectWoo to 1.0.9 and take advantage of data-label support.

### All Submissions:


### Changes proposed in this Pull Request:

Improves A11Y in relation to the billing and shipping state and country fields. Closes #27317.

- Pulls in updated SelectWoo assets, as per [this changeset](https://github.com/woocommerce/selectWoo/pull/43) (which needs to be approved before we can use here).
- Updates our template functions: if we have a label string, we will also add it to the select element as a `data-label` attribute.
- Updates our frontend JS: when creating a SelectWoo dropdown it will look for and utilize the `data-label` string, if provided. 

<img width="976" alt="checkout-forms" src="https://user-images.githubusercontent.com/3594411/115068065-9a92be00-9ea6-11eb-9172-63190ac8539d.png">

☝️ This composite screenshot shows the change in action. Note the tooltips added by MacOS's VoiceOver util, which now is of the format `<Current Value or Placeholder> <Label Text> <Other Items>` whereas, previously, we were missing the `<Label Text>` component (such as _"Country / Region")._

### How to test the changes in this Pull Request:

Using the VoiceOver util from MacOS (of course, feel free to use popular alternatives, too):

1. Start with current trunk code/latest stable release.
    1. Add some items to the cart, proceed to checkout.
    2. Observe the audio feedback (or tooltip text) provided by VoiceOver when you click into our SelectWoo dropdowns (or their labels): the label text itself will not be surfaced.
2. Checkout this code and repeat the process, noting the label text is now discovered by VoiceOver.

### Other information:

* This will be a draft PR until we've evaluated [SelectWoo PR 43](https://github.com/woocommerce/selectWoo/pull/43).
* It looks like we previously bundled SelectWoo 1.0.6, though 1.0.8 is currently the latest tagged version (and the above PR would result in 1.0.9). Going from 1.0.6 to 1.0.9 seems like a safe change, best I can tell, but I wanted to highlight it.
* I tried some lower lift approaches, such as simply nesting the `<select>` (and therefore the generated SelectWoo dropdown) _within_ the `<label>` element, and some other permutations, but besides being inconsistent with how we handle other fields it did not improve things when I tested with VoiceOver.

### Changelog entry

> Enhancement - Improved accessibility of the country and state address fields.
